tests/meltano/core/job/test_job.py

```diff
@@ -89,10 +91,12 @@ async def test_run_failed(self, session):
     @pytest.mark.asyncio
     async def test_run_interrupted(self, session):
         subject = self.sample_job({"original_state": 1}).save(session)
-
         with pytest.raises(KeyboardInterrupt):
             async with subject.run(session):
-                psutil.Process().send_signal(signal.SIGINT)
+                if platform.system() == "Windows":
+                    windows_signal_process(signal.SIGINT)
+                else:
+                    psutil.Process().send_signal(signal.SIGINT)
```

Comment:
What about moving this conditional check and the special handling into the helper function also?

```suggestion
                send_process_signal(signal.SIGINT)
```

And that function can own properly sending the signal depending on the OS check. What do you think?

Reply:
Done, much cleaner!